xarray/tests/test_backends.py

```diff
@@ -1895,20 +1895,23 @@ def test_chunk_encoding_with_dask(self):
             # don't actually check equality because the data could be corrupted
             pass
 
+        # if dask chunks (4) are an integer multiple of zarr chunks (2) it should not fail...
         badenc.var1.encoding["chunks"] = (2,)
-        with pytest.raises(NotImplementedError, match=r"Specified Zarr chunk encoding"):
-            with self.roundtrip(badenc) as actual:
-                pass
+        with self.roundtrip(badenc) as actual:
+            pass
 
+        # if initial dask chunks are aligned, size of last dask chunk doesn't matter
         badenc = badenc.chunk({"x": (3, 3, 6)})
         badenc.var1.encoding["chunks"] = (3,)
-        with pytest.raises(
-            NotImplementedError, match=r"incompatible with this encoding"
-        ):
-            with self.roundtrip(badenc) as actual:
-                pass
+        with self.roundtrip(badenc) as actual:
```

Comment:
Same, let's not use `badenc` when the encoding is not actually bad. May confuse future developers.